src/group_theory/commensurable.lean

```diff
@@ -0,0 +1,161 @@
+import group_theory.index
+import group_theory.quotient_group
+import tactic.linarith
+
+namespace commensurable
+
+
+
+variables {G : Type*} [group G]
+
+def commensurable (H K : subgroup G) : Prop :=
+  subgroup.index (subgroup.subgroup_of H K) ≠ 0 ∧
+  subgroup.index (subgroup.subgroup_of K H) ≠ 0
+
+lemma inf_self_eq_self (H :subgroup G)  : (H ⊓ H).comap H.subtype = ⊤ :=
+begin
+simp only [inf_idem],
+ext,
+simp only [set_like.coe_mem, subgroup.coe_subtype, subgroup.mem_top, iff_self, subgroup.mem_comap],
+end
+
+lemma subgroup_of_top (H : subgroup G) : subgroup.subgroup_of H H = ⊤ :=
+begin
+rw subgroup.subgroup_of,
+have:= inf_self_eq_self H,
+simp at *,
+exact this,
+end
+
+lemma index_of_top  : ( ⊤ : subgroup G).index = 1 :=
+begin
+rw subgroup.index,
+have h3: (cardinal.mk  (quotient_group.quotient (⊤ : subgroup G))) = 1 ,
+  by {rw cardinal.eq_one_iff_unique, split, apply quotient_group.subsingleton_quotient_top,
+    simp only [nonempty_of_inhabited],},
+simp [h3],
+end
+
+lemma reflex (H : subgroup G) : commensurable H H :=
+begin
+rw commensurable,
+simp,
+have:= subgroup_of_top H,
+rw this,
+have h2:= index_of_top,
+rw h2,
+simp,
+end
+
+def conj_subgroup (g : G)  (Γ : subgroup G) : subgroup G :={
+carrier := { h : G | ∃ γ : Γ, g⁻¹ * γ * g = h},
+one_mem' := by  {simp, use 1, simp,}  ,
+mul_mem' := by {intros a b ha hb, simp at *,
+  let aa:= classical.some ha,
+  let bb:= classical.some hb,
+  have haa:= classical.some_spec ha,
+  have hbb:= classical.some_spec hb,
+  use aa*bb, rw   ←  haa, rw ← hbb, simp, simp_rw ←  mul_assoc, simp,},
+inv_mem' := by {intros x hx, simp at *,
+  let a:= classical.some hx,
+  have ha:= classical.some_spec hx,
+  use a⁻¹ ,
+  rw ← ha,
+  simp,
+  simp_rw ←  mul_assoc,} ,
+}
```

Comment:
This is shorter as `mul_aut.conj g
```suggestion
def conj_subgroup (g : G) (Γ : subgroup G) : subgroup G := mul_aut.conj g • Γ
```

Reply:
Oh you got to me before I could ask on Zulip! This is much better, I knew this had to be somewhere in mathlib, thank you!

Reply:
actually it doesnt like the smul, the best I could do was 
```
def conj_subgroup (g : G) (Γ : subgroup G) : subgroup G := 
  subgroup.map (mul_equiv.to_monoid_hom (mul_aut.conj g) )  Γ
```

Reply:
You might need to import subgroup.pointwise

Reply:
Ah yes that worked, nice!